### DISCUSSION TITLE
raudio as upnp server?

### DISCUSSION THREAD

**Discussion by rellimneb** [rern/rAudio#1089] [General]
is it possible to have raudio also run also as a upnp server?   it seems it can only be a upnp client.  i have a drive with all the music attached to a pi running raudio and would sometimes like to access the music from an other device using vlc.

**Comment by EarsofTin** [comment]
Cog wheel settings > Features > enable File Sharing (under Others).
When I do this "rAudio" appears as a local network in VLC.  Then I can cast to various players around the house.

**Reply by rellimneb** [reply]
I have that file sharing option enabled, i can connect to raudio from my mac and copy files but raudio doesn't show up in vlc.  any other options to enable?  i'm using the latest version of raudio on a pi4

**Reply by jtritter** [reply]
Hi, 
sorry for the late answer on this topic, but it seems I'm looking for the same thing as **rellimneb**. 

I have my whole music on my pi3 with touch display using raudio-1. With this device I play my music most time. But I also want to access this music via upnp server from my android tablet and PC (both using VLC). I can't find the raudio in VLC by using "Universal Plug'n Play" - but with activated "File Sharing" I can find it via "mDNS Network" in VLC. But this is just the directory-/file-view using SMB. That's not what I want, as you have to click wild through all directories to find your music. With a "real" upnp server working, you could click directy through mp3-metadata (interpret, album, year, ...).

Is there any upnp-server option in raudio-1? I think the "Upnp renderer" does something else (I just don't understand what :-( ). 
Is it maybe possible to run a miniDLNA-Server in the background - I think this is what volumio does, to solve this problem... (no volumio itself is no solution for me as this system is absolutly horror).

Yours
Jürgen

**Comment by EarsofTin** [comment]
I'm using rAudio 20221014, a thumb drive connected to a RPi3 and the VLC app on an Android tablet. Sometimes I hit refresh in the app to get all sources to appear but usually they appear automatically.
